### PR TITLE
ci: update to Spell check to 0.0.17a

### PR DESCRIPTION
**Pull request by jsoref**
## Summary of the Pull Request
Upgrade the spell checking action

 
## References
https://github.com/microsoft/terminal/pull/9004#issuecomment-771825083

## PR Checklist
* [ ] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### Plurals and paste tenses
In the past, plurals `foo`+`s` and past tenses `foo`+`ed` were automatically tolerated. This turned out to be a bad design choice on my part.

The basic example is that `potatos` would sometimes be treated as a mistake and sometimes not (depending on the presence of `potato`).

You can see in this PR, that this logic resulted in `Applys` being accepted as a word along with `AppContainered` -- there's nothing intrinsically wrong w/ the latter, but unfortunately in order to screen out the former, my shortcut just couldn't stick around. This means that the `dictionary`/`expect` files will grow perhaps by a tiny bit, but as you can see, not really by much.

This is also why `thereses` (a user) was accepted as a word in the past ([therese](https://en.wikipedia.org/wiki/Th%C3%A9r%C3%A8se_of_Lisieux) is in the base dictionary, so `therese` + `s` was acceptable).

### Pull requests

When GitHub initially introduced GitHub Actions, the event for `pull_request` was created without enough permission for a tool like this to work properly. I worked around that by using the `schedule` event. In 2020, they introduced a replacement event `pull_request_target` which has enough permission. This means that I can stop relying on the `schedule` event.

### Miscellaneous

* I've folded together some `expect/` files since now is as good a time as any.
* I've included a hint about `excludes.txt` (I added a similar one for our primary repo recently, and it came up this week in `microsoft/terminal` -- @zadjii-msft)
* I've standardized on a default of `.github/actions/spelling` to make the out of the box experience easier for new adopters, so I'm applying that change here -- if you're attached to the old directory name, specifying it is still supported. -- note the directory rename may cause a merge conflict for people with open PRs and changes to the contents, this shouldn't be a big problem.

## Validation Steps Performed
I've created a temporary repo which is using this as its default branch.

There's a sample PR here:
https://github.com/check-spelling/terminal/pull/1

If people want, they're welcome to create PRs into that repo to test (you can also test in your own forks -- just be sure that you enable or re-enable the workflow -- GitHub has disabled workflows with `schedule` in inactive repositories to save CPU cycles).

### CODE REVIEW

File: src/interactivity/onecore/ConsoleInputThread.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ DWORD WINAPI ConsoleInputThreadProcOneCore(LPVOID /*lpParam*/)
     {
         // If we get an access denied and couldn't connect to the coniosrv in CSRSS.exe.
         // that's OK. We're likely inside an AppContainer in a TAEF /runas:uap test.
-        // We don't want AppContainered things to have access to the hardware devices directly
+        // We don't want things in an AppContainer to have access to the hardware devices directly
```
Comment (DHowett) [inline src/interactivity/onecore/ConsoleInputThread.cpp:92]: okayfine :smile:

**Review by DHowett — APPROVED** [review]
This is great. Thanks so much for keeping spelling maintained & helping us move to each new version. I love the changes in this release. :smile:

**Comment by DHowett** [comment]
I'm guessing the spellbot _currently_ hates this because of the directory rename

**Comment by jsoref** [comment]
Yep.

**Comment by jsoref** [comment]
I suppose I could have done this in two steps:
1. Upgrade the bot w/o the directory rename
2. Rename the directory

I forgot that the current bot would scream at me.

The next version will let you actually [talk to the bot](https://github.com/check-spelling-bot)

And I just fixed a glitch that the sample PR highlighted (the bot knows a thing is misspelled but doesn't give you a cute yellow box for certain squished cases, but since that's just an extra aid and people should aim for 0 comments, it's not a big deal).

**Comment by jsoref** [comment]
For people curious as to why the bot would tattle on the dictionary, it's because it has a different way of reading it.

Namely, there's an entry in the dictionary for:
```
self-abandoningly
```

The bot sees that as `self` and `abandoningly` and it went and looked to see if `abandoningly` was in the dictionary (it isn't), so it tattled. (This is useful because programmers are quite willing to write `foo-bar` as a math/set operation equivalent to `foo - bar`.)

Normally this doesn't happen because the bot is configured to ignore its config files (as @DHowett noted), but here since I renamed the directory, it didn't know to do so. I could have bridged this w/ a PR that added the new directory as an ignore and merged that before doing the rename in a second PR, but ...

And the reason for the second comment is that there's a [force-push](https://github.com/microsoft/terminal/pull/9014#event-4282676104) and the bot managed to spot both instances (one in each of two scheduled hour slots while looking for updated PRs since the last interval it ran).

**Review by zadjii-msft — APPROVED** [review]

**Comment by jsoref** [comment]
@DHowett / @zadjii-msft: maybe I should put the directory name back and restore the config item? otherwise someone will have to merge main into the pending PRs :-(

**Comment by DHowett** [comment]
> @DHowett / @zadjii-msft: maybe I should put the directory name back and restore the config item? otherwise someone will have to merge main into the pending PRs :-(

I’m not bothered by folks having to merge main :)

**Pull request merged**